src/query/storages/fuse/src/io/snapshots.rs

```diff
@@ -226,11 +256,276 @@ impl SnapshotsIO {
         }
     }
 
+    // return from files that last modified time bigger than timestamp
+    #[tracing::instrument(level = "debug", skip_all)]
+    #[async_backtrace::framed]
+    pub async fn get_within_time_files(
```

Comment:
How about using [list_files](https://github.com/datafuselabs/databend/blob/main/src/query/storages/fuse/src/operations/navigate.rs#L232) directly?

Reply:
`get_within_time_files` input a set of files, and filter out files that within time, but `list_files` cannot do that.

Reply:
I mean the fn `list_files` in [navigate.rs](https://github.com/datafuselabs/databend/blob/main/src/query/storages/fuse/src/operations/navigate.rs#L232-L238), add a new arg `prefix`.
```
impl FuseTable {
    #[async_backtrace::framed]
    async fn list_files<P>(&self, time_point: DateTime<Utc>, prefix: String, mut pred: P) -> Result<Vec<String>>
    where P: FnMut(String) {
            let op = self.operator.clone();
            ... ...
     }

        async fn get_orphan_files_to_be_purged(&self, time_point, referenced_files: BTreeSet<String>, ... ) {
               // 1. Filter out all the with in files.
               let files_to_be_purged = self.list_files(time_point, |_| {}).await?;
               // 2. Filter out all the referenced files.
               files_to_be_purged.retain(|location| !referenced_files.contains(location));
               Ok(files_to_be_purged)
        }
}
```